### ISSUE TITLE
Use babel-polyfill

### ISSUE THREAD

**Issue by alidcast** [egoist/babel-preset-vue-app#5]
Is there a reason this setup uses `babel-runtime` over `babel-polyfill`?

As per this issue https://github.com/babel/babel/issues/3825, `babel-polyfill` is most useful for applications, so that it can have as close as possible to a real ES6 environment, since a limitation of 
`babel-runtime` is that it does not work with certain parts of ES6, such as certain instance methods.

**Comment by egoist** [comment]
Thanks for reminding me of the reason why I disabled runtime helpers and polyfills in previous version. So the best solution now is still to disable it and let user handle the polyfill stuffs?

btw, you can still manually include specified polyfills that `babel-plugin-transform-runtime` does not support into your app. Maybe we can add an option to disable runtime helpers and polyfills.

**Comment by alidcast** [comment]
I think it's best to have a smart default with the option of disabling it (or adding to it). 

The reason I opened this issue was because Nuxt uses this plugin for babel presets (https://github.com/nuxt/nuxt.js/issues/271), but since it doesn't provide the original webpack configuration (like Vue templates) it can be cumbersome to modify the entry presets. 

If `babel-polyfill` is recommended for applications, and this plugin is used for app starter templates, what there be a downside of having it as the default? 

I ran into a problem where a plugin that I was creating was using babel `runtime-helpers`,  but when I tried to check if a function was an es6 generator or async function, I started getting errors. I wanted to refrain from using `babel-polyfill` for it, since I saw that it was recommended that they use the runtime helpers and not pollute the global scope. But I will look into providing specific polyfills to `babel-runtime` in the meantime.
